lib/galaxy/tools/actions/upload_common.py

```diff
@@ -282,8 +282,30 @@ def create_paramfile( trans, uploaded_datasets ):
     """
     def _chown( path ):
         try:
-            pwent = pwd.getpwnam( trans.user.email.split('@')[0] )
-            cmd = [ '/usr/bin/sudo', '-E', trans.app.config.external_chown_script, path, pwent[0], str( pwent[3] ) ]
+            # get username from email/username
```

Comment:
There is a lot of duplication here that should be moved to a single function or method called by all who need it.